manifests/apache.pp

```diff
@@ -1,34 +1,122 @@
-        {
-          'path'         => $api_path,
-          'url'          => $api_url,
-          'reverse_urls' => [$api_url],
-        },
-        {
-          'path'         => $content_path,
-          'url'          => $content_url,
-          'reverse_urls' => [$content_url],
-        },
+  $content_base_url = "http://${pulpcore::content_host}:${pulpcore::content_port}"
+  $content_url = "${content_base_url}${content_path}"
+
+  $base_directories = [
+    {
+      provider       => 'directory',
+      path           => $pulpcore::webserver_static_dir,
+      options        => ['Indexes','FollowSymLinks'],
+      allow_override => ['None'],
+    },
+    {
+      'path'            => $content_path,
+      'provider'        => 'location',
+      'request_headers' => [
+        'unset X-CLIENT-CERT',
+        'set X-CLIENT-CERT "%{SSL_CLIENT_CERT}s" env=SSL_CLIENT_CERT',
       ],
+    },
+  ]
+
+  if $pulpcore::remote_user_environ_name {
+    $remote_user_environ_header = $pulpcore::remote_user_environ_name.regsubst(/^HTTP_/, '')
+    $api_directories = [
+      {
+        'path'            => $api_path,
+        'provider'        => 'location',
+        'request_headers' => [
+          "unset ${remote_user_environ_header}",
+          "set ${remote_user_environ_header} \"%{SSL_CLIENT_S_DN_CN}s\" env=SSL_CLIENT_S_DN_CN",
+        ],
+      },
+    ]
+  } else {
+    $api_directories = []
+  }
+
+  $proxy_pass_content = {
+    'path' => $content_path,
+    'url'  => $content_url,
+  }
+
+  $proxy_pass_api = {
+    'path' => $api_path,
+    'url'  => $api_url,
+  }
+
+  case $pulpcore::apache_http_vhost {
+    true: {
+      $http_vhost_name = 'pulpcore'
+      $http_fragment = undef
+
+      include apache
+      include apache::mod::headers
+      apache::vhost { $http_vhost_name:
+        servername  => $pulpcore::servername,
+        port        => $http_port,
+        priority    => $vhost_priority,
+        docroot     => $pulpcore::webserver_static_dir,
+        directories => $base_directories,
+        proxy_pass  => [$proxy_pass_content],
+      }
     }
+    false: {
+      $http_vhost_name = undef
+      $http_fragment = undef
+    }
+    default: {
+      $http_vhost_name = $pulpcore::apache_http_vhost
+      $http_fragment = epp('pulpcore/apache-fragment.epp', {'proxy_pass' => [$proxy_pass_content]})
 
-    if $facts['os']['selinux']['enabled'] {
-      selinux::boolean { 'httpd_can_network_connect': }
     }
   }
+
+  case $pulpcore::apache_https_vhost {
+    true: {
+      $https_vhost_name = 'pulpcore-https'
+      $https_fragment = undef
+
+      include apache
+      include apache::mod::headers
+      apache::vhost { $https_vhost_name:
+        servername  => $pulpcore::servername,
+        port        => $https_port,
+        ssl         => true,
+        priority    => $vhost_priority,
+        docroot     => $pulpcore::webserver_static_dir,
+        directories => $base_directories + $api_directories,
```

Comment:
I think these also need to be passed to the `pulpcore::apache::fragment` use as well so that authentication is configured through remote user environ.